src/plugins/network/network.c

```diff
@@ -50,6 +52,75 @@ int elektraNetworkAddrInfo (Key * toCheck)
 	return 0;
 }
 
+int elektraPortInfo (Key * toCheck, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (toCheck, "check/port");
+	const Key * listenMeta = keyGetMeta (toCheck, "check/port/listen");
+	if (!meta && !listenMeta) return 0; /* No check to do */
+	char * endptr = NULL;
+	long portNumber = strtol (keyString (toCheck), &endptr, 10);
+	int portNumberNetworkByteOrder;
+
+	if (*endptr == '\0')
+	{
+		if (portNumber < 0 || portNumber > 65535)
+		{
+			ELEKTRA_SET_ERRORF (201, parentKey, "Port %ld on key %s was not within 0 - 65535", portNumber, keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = htons (portNumber);
+	}
+	else
+	{
+		struct servent * service;
+		service = getservbyname (keyString (toCheck), NULL); // NULL means we accept both tcp and udp
+		if (service == NULL)
+		{
+			ELEKTRA_SET_ERRORF (202, parentKey, "Could not find service with name %s on key %s", keyString (toCheck),
+					    keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = service->s_port;
+	}
+
+	// Check if we can connect to it
+	if (!listenMeta) return 0; /* No check to do */
+
+	char * hostname = "localhost";
+
+	int sockfd;
+	struct sockaddr_in serv_addr;
+	struct hostent * server;
+	sockfd = socket (AF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0)
+	{
+		ELEKTRA_SET_ERROR (204, parentKey, "Could not open a socket");
+	}
+
+	server = gethostbyname (hostname);
+	if (server == NULL)
+	{
+		ELEKTRA_SET_ERRORF (204, parentKey, "Could not connect to %s: No such host", hostname);
```

Comment:
Is "No such host" the actual reason? The reason is in errno, isn't it?

Btw. we should also handle the TRY_AGAIN case somehow.

Reply:
Fixed errno.

Concerning TRY_AGAIN: Yes we could try to implement a retrying approach but isn't it very unlikely that this happens? Adding this functionality to the code would clearly degrade the readability and maintainability.